src/uct/tcp/tcp_iface.c

```diff
@@ -877,7 +894,9 @@ ucs_status_t uct_tcp_query_devices(uct_md_h md,
     char path_buffer[PATH_MAX];
     ucs_sys_device_t sys_dev;
 
-    n = scandir(UCT_TCP_IFACE_NETDEV_DIR, &entries, NULL, alphasort);
+    n = scandir(UCT_TCP_IFACE_NETDEV_DIR, &entries,
+                tcp_md->config.bridge_devices ? NULL : uct_tcp_filter_bridges,
```

Comment:
it's nice to pass filter function to scandir - but there are other checks below: line 915 and 920-930 which are not in the filter function. So either all should be in the filter function, or none of them

Reply:
Removed filter function.
It is better from code scalability perspective. Because it is not possible to add filter by condition without code duplication.